content/tutorial-plotting-fractals.md

```diff
@@ -0,0 +1,553 @@
+    ax.set_title(f'Mapping of iterations on {cur_val}')
+
+    cycle = ax.scatter(np.real(outputs),np.imag(outputs),c=index,alpha=0.6);
+    fig.colorbar(cycle,label='Iteration');
+```
+
+To our surprise, the behaviour of the function did not come close to matching our hypothesis. This is a prime example of the chaotic behaviour fractals possess. In the first two plots, the value "exploded" on the last iteration, jumping way beyond the region that it was contained in previously. The third plot on the other hand remained bounded to a small region close to the origin, yielding completely different behaviour despite the tiny change in value.
+
+This leads us to an extremely important question: **How many iterations can be applied to each value before they diverge (“explode”)?** 
+
+As we saw from the first two plots, the further the values were from the origin, the faster they generally exploded. Although the behaviour is uncertain for smaller values (like $z_1, z_2, z_3$), we can assume that if a value surpasses a certain distance from the origin (say 2) that it is doomed to diverge. We will call this threshold the **radius**. 
+
+This allows us to quantify the behaviour of the function for a particular value without having to perform as many computations. Once the radius is surpassed, we are allowed to stop iterating, which gives us a way of answering the question we posed. If we tally how many computations were applied before divergence, we gain insight into the behaviour of the function that would be hard to keep track of otherwise.
+
+Of course, we can do much better and design a function that performs the procedure on an entire mesh.
+
+```{code-cell} ipython3
+def divergence_rate(mesh,num_iter=10,radius=2): 
+        
+    z = mesh.copy()
+    diverge_len = np.zeros(mesh.shape) #Keep tally of the number of iterations
+        
+    for i in range(num_iter):
+        #Iterate on element if and only if |element| < radius (Otherwise assume divergence)
+        diverge_len[np.abs(z) < radius] += 1
+        z[np.abs(z) < radius] = f(z[np.abs(z)< radius])   
+        
+        
+    return diverge_len
+```
+
+The behaviour of this function may look confusing at first glance, so it will help to explain some of the notation.
+
+Our goal is to iterate over each value in the mesh and to tally the number of iterations before the value diverges. Since some values will diverge quicker than others, we need a procedure that only iterates over values that have an absolute value that is sufficiently small enough. We also want to stop tallying values once they surpass the radius. For this, we can use **[Boolean Indexing](https://numpy.org/devdocs/reference/arrays.indexing.html#boolean-array-indexing)**, a NumPy procedure that when paired with Universal Functions is unbeatable. Boolean Indexing allows for operations to be performed conditionally on a NumPy array without having to resort to looping over and checking for each array value individually. 
```

Comment:
```suggestion
Our goal is to iterate over each value in the mesh and to tally the number of iterations before the value diverges. Since some values will diverge quicker than others, we need a procedure that only iterates over values that have an absolute value that is sufficiently small enough. We also want to stop tallying values once they surpass the radius. For this, we can use **[Boolean Indexing](https://numpy.org/devdocs/reference/arrays.indexing.html#boolean-array-indexing)**, a NumPy feature that when paired with Universal Functions is unbeatable. Boolean Indexing allows for operations to be performed conditionally on a NumPy array without having to resort to looping over and checking for each array value individually. 
```